### PR TITLE
Add PRO hooks for new lambda provider

### PR DESCRIPTION
**Pull request by joe4dev**
See https://github.com/localstack/localstack-ext/pull/1322

* Add hook for `inject_layer_fetcher`
* Test layers limits (>5 layers)

### CODE REVIEW

File: localstack/services/awslambda/provider.py [hunk 1]
```diff
@@ -467,16 +478,43 @@ def _validate_layers(new_layers: list[str]):
...
+                            "Fetching shared layers from AWS is a pro feature."
+                        )
+
+                    layer = self.layer_fetcher.fetch_layer(layer_version_arn)
+                    if layer is None:
+                        # TODO: detect user or role from context when IAM users are implemented
+                        user = "user/localstack-testing"
+                        raise AccessDeniedException(
+                            f"User: arn:aws:iam::{account_id}:{user} is not authorized to perform: lambda:GetLayerVersion on resource: {layer_version_arn} because no resource-based policy allows the lambda:GetLayerVersion action"
+                        )
```
Comment (dfangl) [inline localstack/services/awslambda/provider.py:516]: I am still not 100% happy with this, as IAM should enforce it, but seems fine to leave it for now.

**Review by dfangl — APPROVED** [review]
LGTM!

File: localstack/services/awslambda/provider.py [hunk 2]
```diff
@@ -187,19 +191,23 @@
...
     router: FunctionUrlRouter
+    layer_fetcher: LayerFetcher | None
 
     def __init__(self) -> None:
         self.lambda_service = LambdaService()
         self.create_fn_lock = threading.RLock()
         self.create_layer_lock = threading.RLock()
         self.router = FunctionUrlRouter(ROUTER, self.lambda_service)
+        self.layer_fetcher = None
+        lambda_hooks.inject_layer_fetcher.run(self)
```
[the remark this answers is not in the document]
Comment (joe4dev) [reply localstack/services/awslambda/provider.py:210]: Agreed. Maybe, I learn more elegant ways to handle this with function plugins in the future ...

File: localstack/services/awslambda/provider.py (hunk 1 above)
Comment (joe4dev) [reply localstack/services/awslambda/provider.py:516]: Ok, maybe better to tackle this in a follow-up ...

**Pull request merged**